docs/MinimallyImpactfulServicing.md

```diff
@@ -0,0 +1,99 @@
+# Minimally Impactful Servicing of eBPF For Windows
+
+## Overview
+As newer versions of eBPF for Windows are released, it is necessary to replace
+the code currently executing on systems. Simply stopping and restarting eBPF is
+not a viable solution as applications are dependent on state stored within eBPF,
+in the form of map, program, and link objects. This document outlines a strategy
+for replacing the currently executing version of eBPF for Windows with a newer
+version.
+
+## Architectural changes
+Under the new architecture, the eBPF for Windows execution context (the portion
+of eBPF hosting the virtual machine in which eBPF programs execute) is split
+into two parts.
+
+The first part hosts the code that interacts with the Windows IO Manager (the
+part of the Windows kernel responsible for exposing handles to user mode). This
+part of the code can not be serviced with out restarting the applications using
+eBPF due to constraints imposed by the Windows kernel. The expectation is that
+this portion of the code will be relatively stable and will not require frequent
+updates. This part is referred to as eBPF Proxy or the proxy driver.
```

Comment:
Why do we need an ebpf proxy, as opposed to using some driver that already exists in Windows?

Reply:
Added section on Potential future improvements to cover this. Currently there is no Windows component that supports this.